api/webhook.go

```diff
@@ -133,6 +140,13 @@ func webhookUpdate(w http.ResponseWriter, r *http.Request, t auth.Token) error {
 	if !permission.Check(t, permission.PermWebhookUpdate, ctx) {
 		return permission.ErrUnauthorized
 	}
+
+	for _, kindName := range webhook.EventFilter.KindNames {
+		if kindName == "webhook.run" {
+			return permission.ErrUnauthorized
```

Comment:
Like in `webhookCreate`, I think unauthorized is not an appropriate response here.